### PR TITLE
More trivial reduction cleanup

### PR DESCRIPTION
**Pull request by zasdfgbnm**
The biggest change is to unify `ConsumerForwardingInfo` and `ProducerForwardingInfo`. Besides that, there are some more misc cleanups.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler/utils.cpp [hunk 1]
```diff
@@ -1451,7 +1450,7 @@ BroadcastMultipleInformation getBroadcastMultiples(
       auto ref_id = ref_root_domain[ref_i];
 
       // If reference id is broadcast or reduction
-      if (ref_id->isBroadcast() || ref_id->isReduction()) {
+      if (ref_id->isBroadcast()) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/scheduler/utils.cpp:1458]: nit: remove "or reduction" above

File: torch/csrc/jit/codegen/cuda/transform_iter.cpp [hunk 2]
```diff
@@ -632,198 +632,112 @@ namespace {
...
-            compliment_ids.push_back(input_id);
-          }
-        }
+    TORCH_INTERNAL_ASSERT(active_root_dom.size() == active_dim_flags->size());
 
-        // Set up compliment map
-        for (auto forwarded_id : forwarded_ids) {
-          compliment_map.emplace(std::make_pair(forwarded_id, compliment_ids));
-        }
+    // Collect which unique broadcast root axes are in only in active_tv but not
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/transform_iter.cpp]: These comments and variable names are a little confusing as they may not about broadcast IDs but squeezed IDs. Maybe just `forwarded_ids`?

**Review by naoyam — APPROVED** [review]
LGTM. Thanks for the cleanup!

File: torch/csrc/jit/codegen/cuda/transform_iter.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/transform_iter.cpp]: changed

File: torch/csrc/jit/codegen/cuda/scheduler/utils.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/scheduler/utils.cpp:1458]: fixed

**Pull request merged**